soccer/gameplay/evaluation/ball.py

```diff
@@ -90,6 +90,8 @@ def angle_btw_three_pts(a, b, vertex):
         VA = math.sqrt((vertex.x - a.x)**2 + (vertex.y - a.y)**2)
         VB = math.sqrt((vertex.x - b.x)**2 + (vertex.y - b.y)**2)
         AB = math.sqrt((a.x - b.x)**2 + (a.y - b.y)**2)
+        if (VA * VA + VB * VB - AB * AB) / (2 * VA * VB) > 1:
```

Comment:
Actually I believe the ValueError is only when the expression results in a number greater than 1, not division by 0, which is what this line of code is meant to handle. It happens when the numerator and denominator are supposed to compute to equal values, but because of programming math the numerator is bigger than the denominator by a miniscule amount and passes in a number slightly larger than 1 to math.acos, which throws the ValueError.